src/cache.h

```diff
@@ -95,7 +95,14 @@ void CacheSystem_init(const char *path, int url_supplied);
 /**
  * \brief clear the content of the cache directory
  */
-void CacheSystem_clear(const char *path);
+void CacheSystem_clear();
+
+/**
+ * \brief Return the fullpath to the cache directory
+ * 
+ * \param[in] cache_or_config 1 to get cache home, 0 for config
+ */
+char *CacheSystem_get_home(int cache_or_config);
```

Comment:
I would rather you rename this ``static char *CacheSystem_get_path(const char *path)``, then have two separate functions calling ``CacheSystem_get_path()``. I don't like the fact that I have to come here to the header file to figure out what 0 and 1 means.

Reply:
Were you expecting that the argument to `CacheSystem_get_path` would now be "config" or "cache" or a name signifying the directory? I made the change so that you need not get to the header to figure out what 0 and 1 means, but  I thought perhaps it would make more sense in future to use an enum instead of the `cache_or_config` being 1 or 0, given there might be other such default directories to be created.

Reply:
I think this function needs rewrite / refactoring. 

With respect to using enum, if it only gets used at one place, then it is probably not worth using enum.